src/renderer/reader/components/ReaderSettings.tsx

```diff
@@ -390,63 +390,65 @@ interface ITable {
 
 const Slider = ({ value, option, set }: { value: string, option: ITable, set: (a: Pick<ReaderConfig, "pageMargins" | "wordSpacing" | "letterSpacing" | "paraSpacing" | "lineHeight">) => void }) => {
     const [currentSliderValue, setCurrentSliderValue] = React.useState(option.defaultValue);
+    const [currentIndex, setCurrentIndex] = React.useState(() => optionsValues[option.parameter].findIndex((el) => el === option.defaultValue));
 
     React.useEffect(() => {
-        setCurrentSliderValue(value.replace(/rem/g, ""));
-    }, [value]);
+        setCurrentSliderValue(value);
+        const newIndex = optionsValues[option.parameter].findIndex((el) => el === value);
+        setCurrentIndex(newIndex);
+    }, [value, option.parameter]);
+
+    const updateValue = (index: number) => {
+        const newValue = optionsValues[option.parameter][index];
+        setCurrentSliderValue(newValue);
+        setCurrentIndex(index);
+        set({ [option.parameter]: newValue } as Pick<ReaderConfig, "pageMargins" | "wordSpacing" | "letterSpacing" | "paraSpacing" | "lineHeight">);
+    };
 
     const click = (direction: string) => {
-        const step = option.step;
-        let newStepValue: number;
-
-        if (direction === "out") {
-            newStepValue = Number(currentSliderValue.replace(/rem/g, "")) - step;
-        } else {
-            newStepValue = Number(currentSliderValue.replace(/rem/g, "")) + step;
-        }
-        const clampedValue = Math.min(Math.max(newStepValue, option.min), option.max);
-        const valueToString = clampedValue.toFixed(2);
-        setCurrentSliderValue(valueToString);
-        set({ [option.parameter]: valueToString + (option.rem ? "rem" : "") } as Pick<ReaderConfig, "pageMargins" | "wordSpacing" | "letterSpacing" | "paraSpacing" | "lineHeight">);
+        setCurrentIndex((prevIndex) => {
+            const newIndex = direction === "out" ? prevIndex - 1 : prevIndex + 1;
+            if (newIndex >= 0 && newIndex < optionsValues[option.parameter].length) {
```

Comment:
Same sanity-check workaround here ... unless we can "guarantee" through the TypeScript type system that `option.parameter` is within bounds of the `optionsValues` array range.

Reply:
As `parameter: keyof AdjustableSettingsStrings` and ```
const optionsValues: AdjustableSettingsStrings = {
    fontSize,
    pageMargins,
    wordSpacing,
    letterSpacing,
    paraSpacing,
    lineHeight,
};
```
option.parameter is guaranteed to be in optionValues array range